src/parser/TextSearchQuery.cpp

```diff
@@ -0,0 +1,166 @@
+      if (configVarToConfigs_[subject.getVariable()].isWordSearch_.value()) {
+        throw TextSearchException(
+            "One search should contain either <contains-word> or "
+            "<contains-entity>.");
+      } else {
+        throw TextSearchException(
+            "One search should only contain one <contains-entity>.");
+      }
+    }
+    configVarToConfigs_[subject.getVariable()].isWordSearch_ = false;
+    if (object.isLiteral()) {
+      std::string_view literal = object.getLiteral().toStringRepresentation();
+      configVarToConfigs_[subject.getVariable()].entity_ =
+          std::string(literal.substr(1, literal.size() - 2));
+    } else if (object.isVariable()) {
+      configVarToConfigs_[subject.getVariable()].entity_ = object.getVariable();
+    } else {
+      throw TextSearchException(
+          "The predicate <contains-entity> needs a Variable as subject and a "
+          "Literal or Variable as object.");
+    }
+  } else if (predString == "bind-match") {
+    throwVariableException();
+    if (configVarToConfigs_[subject.getVariable()]
+            .varToBindMatch_.has_value()) {
+      throw TextSearchException(
+          "A text search should only contain one <bind-match>.");
+    }
+    configVarToConfigs_[subject.getVariable()].varToBindMatch_ =
+        object.getVariable();
+  } else if (predString == "bind-score") {
+    throwVariableException();
+    if (configVarToConfigs_[subject.getVariable()]
+            .varToBindScore_.has_value()) {
+      throw TextSearchException(
+          "A text search should only contain one <bind-score>.");
+    }
+    configVarToConfigs_[subject.getVariable()].varToBindScore_ =
+        object.getVariable();
+  }
+}
+
+// ____________________________________________________________________________
+std::vector<std::variant<TextIndexScanForWordConfiguration,
+                         TextIndexScanForEntityConfiguration>>
+TextSearchQuery::toConfigs(QueryExecutionContext* qec) const {
+  std::vector<std::variant<TextIndexScanForWordConfiguration,
+                           TextIndexScanForEntityConfiguration>>
+      output;
+  // First pass to get all word searches
+  ad_utility::HashMap<Variable, vector<string>> potentialTermsForCvar;
+  for (const auto& [var, conf] : configVarToConfigs_) {
+    if (!conf.isWordSearch_.has_value()) {
+      throw TextSearchException(
+          "Text search service needs either <contains-word> or "
+          "<contains-entity>.");
+    }
+    if (!conf.textVar_.has_value()) {
+      throw TextSearchException(
+          "Text search service needs a text variable to search.");
+    }
+    if (conf.isWordSearch_.value()) {
+      auto it = potentialTermsForCvar.find(conf.textVar_.value());
+      if (it == potentialTermsForCvar.end()) {
+        potentialTermsForCvar.insert(
+            {conf.textVar_.value(), {conf.word_.value()}});
+      } else {
+        it->second.push_back(conf.word_.value());
+      }
```

Comment:
```suggestion
      potentialTerms[textVar].push_back(word);
```
easy as that, the `[]` automatically inserts an empty vector.

Reply:
5761b223dcab23b580ea74da0732cbb31348b8cc